### PR TITLE
[READY] Vendors can tip on people thrown at them extra hard, lets you set ejection speed on disposal outlets

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
This PR gives people who are thrown extra hard into vending machines a chance to be crushed by said vending machine. Being thrown extra hard means being thrown at a speed higher than your normal throw_speed (for carbons, 2). Here's a quick list of throw_speed mods for carbons.

- Thrown by a hulk: +1 speed
- Throwing a dwarf: +1 speed
- Thrown from a neckgrab: +1 speed
- Thrown by a dwarf: -1 speed

Each extra throw_speed adds 20% to your chance to get squished, so being thrown by a hulk gives you a 20% chance, being neckthrown by a hulk gives you a 40% chance, and being a dwarf thrown by a hulk who has you in a neckgrab gives you a 60% chance. This PR also lets you use a multitool on a disposal outlet to change the eject speed between 1/2/4 rather than just 1. If you emag it, you can crank it up to 6! Fun!

There's other things that can throw you faster like baseball bats (40% of the time you launch at 4 speed) and cartoonish sneezing (random from 1-4 speed), so experiment! 

## Why It's Good For The Game
Adds more ways to use the environment to your advantage in a battle, gives hulk a small yet flavorful buff that isn't just "click to smash thing"

## Changelog
:cl: Ryll/Shaps
add: Throwing people really hard into vending machines is not advised if you don't want to clean them up afterwards. Things are considered thrown really hard if: You're throwing them as a hulk, the person you're throwing is a dwarf, and/or if you're throwing someone from a neckgrab
add: You can now adjust the speed disposal outlets eject contents at with a multitool, and emag the disposal outlet to amp the speed up even further!
/:cl:

### CODE REVIEW

**Comment by Shadowflame909** [comment]
I could have swarn something like this existed.

Was probably just me skateboarding into a vending machine

File: code/modules/recycling/disposal/outlet.dm [hunk 1]
```diff
@@ -80,3 +84,39 @@
...
+		if(EJECT_SPEED_YEET)
+			eject_speed = EJECT_SPEED_SLOW
+	return TRUE
+
+/obj/structure/disposaloutlet/emag_act(mob/user, obj/item/card/emag/E)
+	. = ..()
+	if(emagged)
+		return
+	to_chat(user, "<span class='notice'>You silently disable the sanity checking on \the [src]'s ejection force.</span>")
+	emagged = TRUE
```
Comment (ATH1909) [inline code/modules/recycling/disposal/outlet.dm:122]: isn't there a flag for this?

File: code/datums/components/tackle.dm [hunk 2]
```diff
@@ -434,13 +435,13 @@
 		W.obj_destruction()
 		user.adjustStaminaLoss(10 * speed)
 		user.Paralyze(30)
-		user.visible_message("<span class='danger'>[user] slams into [W] and shatters it, shredding [user.p_them()]self with glass!</span>", "<span class='userdanger'>You slam into [W] and shatter it, shredding yourself with glass!</span>")
+		user.visible_message("<span class='danger'>[user] smacks into [W] and shatters it, shredding [user.p_them()]self with glass!</span>", "<span class='userdanger'>You smacks into [W] and shatter it, shredding yourself with glass!</span>")
 
 	else
-		user.visible_message("<span class='danger'>[user] slams into [W] like a bug, then slowly slides off it!</span>", "<span class='userdanger'>You slam into [W] like a bug, then slowly slide off it!</span>")
```
Comment (ATH1909) [inline code/datums/components/tackle.dm:440]: aw, I liked the "slowly slides off it" part...

**Comment by Fabian272** [comment]
Bones go crunch

**Comment by ArcaneDefence** [comment]
Having seen a group of players build a trampoline out of shield generators that would ping-pong people across several screens of space to then see the mime put a vending machine in the way deliberately, I look forward to the levels of shenanigans this causes.

**Comment by MrDoomBringer** [comment]
absolutely blessed 
why is it that i can always tell a shaps PR from the title alone

File: code/modules/recycling/disposal/outlet.dm (hunk 1 above)
Comment (kriskog) [reply code/modules/recycling/disposal/outlet.dm:122]: There is, its in obj_flags

Comment (Ryll-Ryll) [reply code/modules/recycling/disposal/outlet.dm]: fixed

**Review by kriskog — APPROVED** [review]

**Pull request merged**